src/macros/macrorecorder.cpp

```diff
@@ -0,0 +1,106 @@
+#include "macrorecorder.h"
+
+#include <QtConcurrentRun>
+
+#include "control/controlproxy.h"
+#include "preferences/configobject.h"
+
+namespace {
+constexpr uint kMaxMacroSize = 1000;
+const QString kConfigGroup = QStringLiteral("[MacroRecording]");
+}
+
+MacroRecorder::MacroRecorder()
+        : m_COToggleRecording(ConfigKey(kConfigGroup, "recording_toggle")),
```

Comment:
In Mixxx we have a toggle button. Take a look at Play or or [Shoutcast], enable 
https://github.com/mixxxdj/mixxx/blob/52c9145f3979d12866b0ee3e1a23eb19b72fb272/src/broadcast/broadcastmanager.cpp#L29

Adopted to this use case, I think it is toggle button "[MacroRecording], record"

Reply:
Ah, nice. I think I copied that from RecordingManager, so maybe we should change it there as well.